lib/puppet/provider/group/groupadd.rb

```diff
@@ -17,11 +17,20 @@
     value.is_a? Integer
   end
 
-  optional_commands :localadd => "lgroupadd", :localdelete => "lgroupdel", :localmodify => "lgroupmod"
-
-  has_feature :manages_local_users_and_groups, :manages_members if Puppet.features.libuser?
-
-  options :members, :flag => '-M', :method => :mem
+  optional_commands :localadd => "lgroupadd", :localdelete => "lgroupdel", :localmodify => "lgroupmod", :purgemember => "usermod"
+
+  has_feature :manages_local_users_and_groups if Puppet.features.libuser?
+  has_feature :manages_members if Puppet.features.libuser? ||
+                                  (Puppet.runtime[:facter].value('os.name') == "Fedora" &&
```

Comment:
Is there a better way to determine if the feature is missing then just going by OS? Right now this seems sane with just Fedora 40, but when upstream distros start to implement this will the logic be as cut and dry?

Reply:
I was also hesitant to go with this specific OS condition but had to because I couldn't find any other way. 

I tried to condition it based on the `Puppet.features.libuser?` condition but there are other platforms like Debian that don't have libuser and cannot manage members, but Fedora also doesn't have libuser but can manage members. So, doing so would impact Debian atleast.

I think we may have to go with approach for now, but if anyone has any suggestions please let me know.

Reply:
It sounds like Fedora 40 is able to manage group members without libuser because `groupmod` was updated sometime recently to include the functionality. Would we be able to identify which version of `shadow-utils` in which that functionality was added and check for that?